src/app/src/components/ConfirmActionButton.jsx

```diff
@@ -0,0 +1,231 @@
+import React, { Component } from 'react';
+
+import { arrayOf, bool, func, shape, string } from 'prop-types';
+
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+
+import { CONFIRM_ACTION, MERGE_ACTION, REJECT_ACTION } from '../util/constants';
+
+const actionDialogStates = Object.freeze({
+    none: 'none',
+    confirm: CONFIRM_ACTION,
+    merge: MERGE_ACTION,
+    reject: REJECT_ACTION,
+});
+
+export default class ConfirmActionButton extends Component {
```

Comment:
1. According to the [latest recommendation](https://react.dev/reference/react/Component) from the React documentation, it's advisable to rewrite all class components as functional ones. My goal is to exclusively use functional components because the use of componentDidMount(), componentWillUnmount(), etc., is considered outdated.

E.g.
```
const MyComponent = ({prop1, prop2}) => {
......
return ();
}
```

2. Could you please use named export?

Default exports don’t associate any name with the item being exported, meaning that any name can be applied during import. This may give you a flexibility, but if multiple developers imports the same module with different names or even non descriptive name, we are screwed.
Named exports are explicit, forcing the consumer to import with the names the original author intended and removing any ambiguity.

E.g.

```
// ❌
export default MyComponent;

// ✅
export { MyComponent };
```

Reply:
> it's advisable to rewrite all class components as functional ones

This is a topic for tech debt since we have ~30 classes in the FE app.

> Could you please use named export?

Same

Reply:
> it's advisable to rewrite all class components as functional ones

I mean only the classes you created.

Reply:
Could you please describe the reasons why this component cannot be functional?